catt/controllers.py

```diff
@@ -92,50 +63,81 @@ def setup_cast(device_name, video_url=None, prep=None, controller=None, ytdl_opt
     except (pychromecast.error.ChromecastConnectionError, ValueError):
         cast = get_chromecast(device_name)
         cache.set_data(cast.name, cast.host, cast.port)
+
     cast.wait()
+    return cast
 
-    if video_url:
-        model_name = DEVICES_WITH_TWO_MODEL_NAMES.get(cast.model_name, cast.model_name)
-        cc_info = (cast.device.manufacturer, model_name)
-        stream = StreamInfo(
-            video_url, model=cc_info, host=cast.host, device_type=cast.cast_type, ytdl_options=ytdl_options
-        )
 
-    if controller:
-        if controller == "default":
-            app = DEFAULT_APP
-        else:
-            app = next(a for a in APP_INFO if a["app_name"] == controller)
-    elif stream and prep == "app":
-        if stream.is_local_file:
-            app = DEFAULT_APP
+def get_stream(url, device_info=None, host=None, ytdl_options=None):
+    if device_info:
+        model_name = DEVICES_WITH_TWO_MODEL_NAMES.get(device_info.model_name, device_info.model_name)
+        cc_info = (device_info.manufacturer, model_name)
+        cast_type = device_info.cast_type
+    else:
+        cc_info = cast_type = None
+    return StreamInfo(url, host=host, model=cc_info, device_type=cast_type, ytdl_options=ytdl_options)
+
+
+def get_app_info(id_or_name, cast_type=None, strict=False, show_warning=False):
+    if id_or_name == "default":
+        return DEFAULT_APP
+
+    field = "app_id" if re.match("[0-9A-F]{8}$", id_or_name) else "app_name"
+    try:
+        app_info = next(a for a in APP_INFO if a[field] == id_or_name)
+    except StopIteration:
+        if strict:
+            raise AppSelectionError("app not found (strict is set)")
```

Comment:
```suggestion
            raise AppSelectionError("App not found (strict is set)")
```

Reply:
I'm a bit uncertain on which case to use with exception messages that are not the ```click``` kind. Looking at exceptions from other contexts, additional messages seem to (mostly) start with lower-case.

Reply:
Hm yeah. I generally prefer starting with uppercase, but I guess we can go either way, that's fine.